stackit/internal/services/ske/cluster/resource.go

```diff
@@ -1323,7 +1323,11 @@ func mapNodePools(ctx context.Context, cl *ske.Cluster, m *Model) error {
 			nodePool["cri"] = types.StringPointerValue(nodePoolResp.Cri.Name)
 		}
 
-		err := mapTaints(nodePoolResp.Taints, nodePool)
+		modelNodePoolTaints := types.ListNull(types.ObjectType{AttrTypes: taintTypes})
+		if i < len(modelNodePools) {
+			modelNodePoolTaints = modelNodePools[i].Taints
+		}
```

Comment:
Can you explain what this does? I'm not sure I understand

Reply:
`i` is the index of the nodepool we are mapping. If `i < len(modelNodePools)` then that node pool is already in the model (we are in a create or update). Otherwise we can be are in a read (e.g. an import or a data source, where the model was previously `nil`)

Reply:
But maybe there is a better was to do it

Reply:
Ok I get it now., thanks!

But then even if we are in a read operation, the model should have a Taints property that would be Null or Unknown. 
I think the mapTaints function should receive the contents of the model directly and then do the mapping internally depending on the current state. This could be achieved with an if `.IsUnknown` or `.IsNull`. Right? This external mapping is breaking the separation of concerns and I don't think there is any gain

Reply:
Adjusted the code. I think the logic to _map_ should be kept be in the `mapTaints` function, so I am just passing a bool to indicate if the current node pool model has anything in the taints field or not (not meaning that is either null or unknown)